hedyorg/hedy
# [UI] Enable alternative error captions for a testing group

_PR DRAFT of the Köln students :)_

**Description**

_At every `/parse` request that leads to a `HedyExeption`, the users receive an alternative error caption, if they are in the test group **AND** the used language is a supported test language.
If the user have no `test_group` field in their DB entry, it will be created. (1/3 users with supported languages will be selected.)_

**Fixes _#3191_**

**How to test**

* Login with a user of your choice in a local env.
* Change language to _En_ or _De_ (if it isn't already).
* Go to Hedy application and Click on "Run code".
* You should now find a field `test_group` within your user entry with `True` or `False` in `./dev_database.json`.
* If `test_group` is `False`, set it to `True`.
* In the code section of the Hedy app just type `echo` without anything to provoke a error.
* You now should see the alternative error caption `B_TEST: We can't run your program.` (Still a template.)
* If it doesn't appears, clear your browser cache and Hedy cookies. 

_We will find some better alternatives for the transpile error caption in the next steps and check how to archive and analyze the test data._
